src/docs/arch-cursor.dox

```diff
@@ -3,13 +3,393 @@
+cursor indicated by the numeric argument will be used to get keys or values that follow.
+The 'n' action gets one or more next key or value columns from the cursor. The 's' skips
+one or more columns. Switching to another cursor resets the cursor to point at the first
+column in the indicated cursor.  This allows a way to find keys and values in arbitrary
+order.
+
+As a contrived example, suppose we have a table with two keys
+\c "k1,k2" and four values \c "x1,x2,x3,x4".  The column groups are created as follows:
+```
+    session->create(session, "colgroup:main:c1", "columns=(x1,x2)");
+    session->create(session, "colgroup:main:c2", "columns=(x3,x4)");
+```
+The column group \c "main:c1" has keys "k1,k2" and values "x1,x2";
+the column group \c "main:c2" has keys "k1,k2" and values "x3,x4".  A table cursor
+on the main table will have two sub-cursors, one for each column group.
+Now consider the ill-considered projection that is opened as:
+```
+    session->open_cursor(session, "table:main(x1,x4,x3,k2,k1,x2,x4)", NULL, NULL, &cursor);
+
+```
+In this case, the plan is this string (with spacing added):
+```
+0v n 1v s n 1v n 0k s n 0k n 0v s n 1v s n
+```
+To break it down,   
+- \c "0v" means use sub-cursor 0 and prepare to read values.
+- \c "n" means get the next value, that is \c "x1".
+- \c "1v" means use sub-cursor 1 and prepare to read values.
+- \c "s" means skip a value, that is \c "x3".
+- \c "n" means get the next value, that is \c "x4".
+- \c ...
+
+Notice that this plan requires many switches of cursors and several \c "s" (skip) operations.
+Each skip involves enough decoding of the data item to determine its length so its data can
+be skipped over.
+
+With the default (complete) projection, getting values is fast.  Using the default plan, the needed
+columns are pulled out of each subordinate cursor one by one, being copied to the caller's arguments.
+With projections, the simple algorithm following the plan works well if the columns in the projection
+are grouped by column group and requested in order.  Without that discipline, as in this example,
+the performance will not be optimal.
+
+The implementation of plan creation and execution resides the schema.
+
+@section arch_cursor_dump Dump cursors
+
+Dump cursors are used in two rather different ways.  Regular dump cursors retrieve the raw keys and values
+and translate bytes either as raw characters or translating to hex.  This flavor of dump cursor is used
+by the `wt dump` or `wt dump -x` utility.  JSON dump cursors do more sophisticated translation, returning a string that is a JSON
+formatted record, with the name for each key and its data and the name for each value and its corresponding data.
+Data is translated to either integral, floating point, or string depending on the format of the column in the schema.
+This flavor of dump cursor is used by the `wt dump -j` command as part of creating a JSON dump of WiredTiger URI.
```

Comment:
of **_the_** WiredTiger URI

Reply:
Changed this to `JSON dump of a WiredTiger table or file`, I hope that's clearer.